Sources/PortalSwift/Portal.swift

```diff
@@ -99,168 +197,458 @@ public class Portal {
       featureFlags: featureFlags
     )
 
-    // Initialize the Portal API
-    self.api = PortalApi(apiKey: apiKey, apiHost: apiHost, provider: self.provider, featureFlags: self.featureFlags)
+    // Retain backward compatible chainId behavior
+    self.provider.chainId = chainId
 
-    // Ensure storage adapters have access to the Portal API
-    if backup.gdrive != nil {
-      backup.gdrive?.api = self.api
-    }
-    if backup.icloud != nil {
-      backup.icloud?.api = self.api
+    // Initialize the Portal API
+    let api = PortalApi(apiKey: apiKey, apiHost: apiHost, provider: self.provider, featureFlags: self.featureFlags)
+    self.api = api
+    keychain.api = api
+
+    // This is to mimic the blocking behavior of the legacy GetClient() implementation
+    // It ensures address information is available at the completion of the initialization
+    let semaphore = DispatchSemaphore(value: 0)
+    Task {
+      // Load client metadata
+      try await keychain.loadMetadata()
+      semaphore.signal()
     }
+    semaphore.wait()
+    // End legacy GetClient() implementation
 
-    if #available(iOS 16, *) {
-      if backup.passkeyStorage != nil {
-        backup.passkeyStorage?.portalApi = self.api
-        backup.passkeyStorage?.apiKey = self.apiKey
-      }
-    }
+    print("Portal initializer done!")
 
     // Initialize Mpc
     self.mpc = PortalMpc(
       apiKey: apiKey,
       api: self.api,
       keychain: keychain,
-      storage: backup,
-      isSimulator: isSimulator,
       host: mpcHost,
+      isSimulator: isSimulator,
       version: version,
       mobile: self.binary,
       apiHost: self.apiHost,
       featureFlags: self.featureFlags
     )
 
+    // Ensure storage adapters have access to the Portal API
```

Comment:
Using the new `registerBackupMethod` convention to initialize all backup items from the provided configuration.